src/components/editMeasure/populationCriteria/baseConfiguration/BaseConfiguration.test.tsx

```diff
@@ -0,0 +1,291 @@
+import * as React from "react";
+import {
+  render,
+  fireEvent,
+  waitFor,
+  screen,
+  within,
+} from "@testing-library/react";
+import { act } from "react-dom/test-utils";
+import BaseConfiguration from "./BaseConfiguration";
+import useMeasureServiceApi, {
+  MeasureServiceApi,
+} from "../../../../api/useMeasureServiceApi";
+import { Measure } from "@madie/madie-models";
+import userEvent from "@testing-library/user-event";
+import { checkUserCanEdit, measureStore } from "@madie/madie-util";
+
+const mockHistoryPush = jest.fn();
+
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useHistory: () => ({
+    push: mockHistoryPush,
+  }),
+}));
+
+jest.mock("../../../../api/useMeasureServiceApi");
+const measure = {
+  id: "test measure",
+  measureName: "the measure for testing",
+  cqlLibraryName: "TestCqlLibraryName",
+  ecqmTitle: "ecqmTitle",
+  measurementPeriodStart: "01/01/2022",
+  measurementPeriodEnd: "12/02/2022",
+  createdBy: "john doe",
+  measureSetId: "testMeasureId",
+  acls: [{ userId: "othertestuser@example.com", roles: ["SHARED_WITH"] }], //#nosec
+} as unknown as Measure;
+
+jest.mock("@madie/madie-editor", () => ({
+  synchingEditorCqlContent: jest.fn().mockResolvedValue("modified cql"),
+  parseContent: jest.fn(() => []),
+  validateContent: jest.fn().mockResolvedValue({
+    errors: [],
+    translation: { library: "NewLibName" },
+  }),
+}));
+
+jest.mock("@madie/madie-util", () => ({
+  useOktaTokens: jest.fn(() => ({
+    getAccessToken: () => "test.jwt",
+  })),
+  useKeyPress: jest.fn(() => false),
+  measureStore: {
+    updateMeasure: jest.fn(),
+    state: jest.fn(),
+    initialState: jest.fn(),
+    subscribe: (set) => {
+      return { unsubscribe: () => null };
+    },
+  },
+  routeHandlerStore: {
+    subscribe: (set) => {
+      set();
+      return { unsubscribe: () => null };
+    },
+    updateRouteHandlerState: () => null,
+    state: { canTravel: true, pendingPath: "" },
+    initialState: { canTravel: true, pendingPath: "" },
+  },
+
+  checkUserCanEdit: jest.fn(() => {
+    return true;
+  }),
+}));
+
+const useMeasureServiceApiMock =
+  useMeasureServiceApi as jest.Mock<MeasureServiceApi>;
+
+let serviceApiMock: MeasureServiceApi;
+
+describe("MeasureInformation component", () => {
+  const { getByTestId, findByTestId, findAllByTestId, getByText } = screen;
+
+  test("Measure Group Scoring renders to correct options length, and defaults to empty string", async () => {
+    measureStore.state.mockImplementationOnce(() => measure);
+    checkUserCanEdit.mockImplementationOnce(() => true);
```

Comment:
Looks like all test cases are mocking checkUserCanEdit and returning true, Feels redundant unless a negative test case has been missing.

Reply:
The same goes for ` measureStore.state.mockImplementationOnce(() => measure);`

Reply:
I removed it to beforeEach.

Reply:
If all test cases are going to use the same return value, Then for checkUserCanEdit on line 71, The module is already mocked and it is returning "true", so I believe it is not required anymore in beforeEach. 

For measureStore.state on line 55 you are returning jest.fn() instead you can use the "measure" object so that the entire test suite can use the same measure object.

Reply:
Ok this is changed. Thanks Rohit!